pkg/client/src/app/api/models.tsx

```diff
@@ -342,9 +342,9 @@ export interface TaskData {
     withDeps: boolean;
     artifact: string;
   };
-  targets: string[];
-  sources: string[];
-  scope: {
+  targets?: string[];
+  sources?: string[];
+  scope?: {
```

Comment:
I don't think this is needed. Actually the backend has those fields mandatory.
* targets cannot be empty.
* sources can be empty but it's an empty array in that case.
* scope is mandatory where withKnown is false by default and packages.includes = [] by default.